--- navit/gui/gtk/gui_gtk_statusbar.c.orig	2010-05-24 01:19:24.000000000 -0400
+++ navit/gui/gtk/gui_gtk_statusbar.c	2010-05-24 01:20:31.000000000 -0400
@@ -166,7 +166,7 @@
 	gtk_box_pack_start(GTK_BOX(this->hbox), this->gps, TRUE, TRUE, 2);
 	gtk_box_pack_start(GTK_BOX(this->hbox), gtk_vseparator_new(), TRUE, TRUE, 2);
 	gtk_box_pack_start(GTK_BOX(this->hbox), this->route, TRUE, TRUE, 2);
-	GTK_WIDGET_UNSET_FLAGS (this->hbox, GTK_CAN_FOCUS);
+	gtk_widget_set_can_focus(this->hbox, FALSE);
 
 	gtk_box_pack_end(GTK_BOX(gui->vbox), this->hbox, FALSE, FALSE, 0);
 	gtk_widget_show_all(this->hbox);
