diff -Ncr maliit-maliit-framework_orig/src/src.pro maliit-maliit-framework/src/src.pro
*** maliit-maliit-framework_orig/src/src.pro	2011-09-21 23:50:24.000000000 +0900
--- maliit-maliit-framework/src/src.pro	2011-10-07 12:05:03.253907424 +0900
***************
*** 166,176 ****
      gconf_config_source = $$system(gconftool-2 --get-default-source)
  }
  
- QMAKE_EXTRA_TARGETS += register_schemas
- register_schemas.target = register_schemas
- register_schemas.commands += GCONF_CONFIG_SOURCE=$$gconf_config_source $$gconftool --makefile-install-rule $$install_schemas.files
- install_schemas.depends += register_schemas
- 
  # Check targets
  QMAKE_EXTRA_TARGETS += check-xml
  check-xml.target = check-xml
--- 166,171 ----
