--- clients/Makefile.am.bak	2010-10-18 21:35:01.000000000 -0400
+++ clients/Makefile.am	2010-10-18 21:05:34.000000000 -0400
@@ -12,12 +12,13 @@
 	window.c				\
 	window.h				\
 	wayland-glib.c				\
 	wayland-glib.h				\
 	cairo-util.c				\
 	cairo-util.h
+libtoytoolkit_la_LIBADD = $(top_builddir)/wayland/libwayland-client.la
 
 flower_SOURCES = flower.c
 gears_SOURCES = gears.c
 screenshot_SOURCES = screenshot.c screenshooter-protocol.c
 terminal_SOURCES = terminal.c
 image_SOURCES = image.c

